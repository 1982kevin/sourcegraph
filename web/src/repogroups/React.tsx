import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
