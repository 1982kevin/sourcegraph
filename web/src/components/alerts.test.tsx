import { ErrorAlert } from './alerts'
import React from 'react'
import { createMemoryHistory } from 'history'
import { mount } from 'enzyme'

jest.mock('mdi-react/AlertCircleIcon', () => 'AlertCircleIcon')

describe('ErrorAlert', () => {
    it('should render an Error object as an alert', () => {
        expect(
            mount(<ErrorAlert error={new Error('an error happened')} history={createMemoryHistory()} />)
        ).toMatchSnapshot()
    })

    it('should add a prefix if given', () => {
        expect(
            mount(
                <ErrorAlert
                    error={new Error('an error happened')}
                    prefix="An error happened"
                    history={createMemoryHistory()}
                />
            )
        ).toMatchSnapshot()
    })

    it('should omit the icon if icon={false}', () => {
        expect(
            mount(<ErrorAlert error={new Error('an error happened')} icon={false} history={createMemoryHistory()} />)
        ).toMatchSnapshot()
    })

    it('should render a Go multierror nicely', () => {
        expect(
            mount(
                <ErrorAlert
                    error={
                        new Error(
                            '- Additional property asdasd is not allowed\n- projectQuery.0: String length must be greater than or equal to 1\n'
                        )
                    }
                    history={createMemoryHistory()}
                />
            )
        ).toMatchSnapshot()
    })
})
