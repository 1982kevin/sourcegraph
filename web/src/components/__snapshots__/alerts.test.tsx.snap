// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert should add a prefix if given 1`] = `
<ErrorAlert
  error={[Error: an error happened]}
  history="[History]"
  prefix="An error happened"
>
  <div
    className="alert alert-danger"
  >
    <AlertCircleIcon
      className="icon icon-inline"
    />
     
    <strong>
      An error happened
      :
    </strong>
     
    <ErrorMessage
      error={[Error: an error happened]}
      history="[History]"
    >
      <Markdown
        dangerousInnerHTML="An error happened"
        history="[History]"
        wrapper="span"
      >
        <span
          className="markdown"
          dangerouslySetInnerHTML={
            Object {
              "__html": "An error happened",
            }
          }
          onClick={[Function]}
        />
      </Markdown>
    </ErrorMessage>
  </div>
</ErrorAlert>
`;

exports[`ErrorAlert should omit the icon if icon={false} 1`] = `
<ErrorAlert
  error={[Error: an error happened]}
  history="[History]"
  icon={false}
>
  <div
    className="alert alert-danger"
  >
     
     
    <ErrorMessage
      error={[Error: an error happened]}
      history="[History]"
    >
      <Markdown
        dangerousInnerHTML="An error happened"
        history="[History]"
        wrapper="span"
      >
        <span
          className="markdown"
          dangerouslySetInnerHTML={
            Object {
              "__html": "An error happened",
            }
          }
          onClick={[Function]}
        />
      </Markdown>
    </ErrorMessage>
  </div>
</ErrorAlert>
`;

exports[`ErrorAlert should render a Go multierror nicely 1`] = `
<ErrorAlert
  error={
    [Error: - Additional property asdasd is not allowed
- projectQuery.0: String length must be greater than or equal to 1
]
  }
  history="[History]"
>
  <div
    className="alert alert-danger"
  >
    <AlertCircleIcon
      className="icon icon-inline"
    />
     
     
    <ErrorMessage
      error={
        [Error: - Additional property asdasd is not allowed
- projectQuery.0: String length must be greater than or equal to 1
]
      }
      history="[History]"
    >
      <Markdown
        dangerousInnerHTML="<ul>
<li>Additional property asdasd is not allowed</li>
<li>projectQuery.0: String length must be greater than or equal to 1</li>
</ul>"
        history="[History]"
        wrapper="span"
      >
        <span
          className="markdown"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<ul>
<li>Additional property asdasd is not allowed</li>
<li>projectQuery.0: String length must be greater than or equal to 1</li>
</ul>",
            }
          }
          onClick={[Function]}
        />
      </Markdown>
    </ErrorMessage>
  </div>
</ErrorAlert>
`;

exports[`ErrorAlert should render an Error object as an alert 1`] = `
<ErrorAlert
  error={[Error: an error happened]}
  history="[History]"
>
  <div
    className="alert alert-danger"
  >
    <AlertCircleIcon
      className="icon icon-inline"
    />
     
     
    <ErrorMessage
      error={[Error: an error happened]}
      history="[History]"
    >
      <Markdown
        dangerousInnerHTML="An error happened"
        history="[History]"
        wrapper="span"
      >
        <span
          className="markdown"
          dangerouslySetInnerHTML={
            Object {
              "__html": "An error happened",
            }
          }
          onClick={[Function]}
        />
      </Markdown>
    </ErrorMessage>
  </div>
</ErrorAlert>
`;
